### PR TITLE
Add the ability for `mailmason` to generate and upload/update templates.

### PR DESCRIPTION
**Pull request by randytarampi**
My take on #1 , but built in a less standalone way and more to support a mailmason centric workflow – this is coupled heavily/ a prerequisite with wildbit/mailmason#34.

I've reworked a couple of things here:
- Rejigged the configuration files to be more in-line with mailmason, just for my own convenience copying and pasting secrets
- Extracted the template file name to the configuration
- `postmark-templates` becomes `postmark-templates-upload`
- `postmark-templates-upload` is now a task that reads some `HtmlBody` and `TextBody` from the templates file instead of reading a file specified in the target configuration.
  - This is done to keep `postmark-templates-output` as dumb as possible, and to facilitate an easier, more transparent coupling with `mailmason` (via a proposed `postmark-templates-generate` task).
  - To keep the `templates.json` file clean, you can specify a `config.templates.clean_output` option which will not write the `HtmlBody` and `TextBody` back out.

### CODE REVIEW

**Comment by randytarampi** [comment]
Would be great to hear what @hybernaut thinks of this. Couldn't wait on #1 to be merged so I had to roll my own solution for now, though I'll gladly move to whatever goes onto `master`.

**Comment by randytarampi** [comment]
@derekrushforth @hybernaut Any comments on this or wildbit/mailmason#34?

File: example_config.json
Comment (derekrushforth) [inline example_config.json]: Let's remove this step since grunt-postmark isn't dependent on MailMason.

File: example_secrets.json
Comment (derekrushforth) [inline example_secrets.json]: Let's remove this step since grunt-postmark isn't dependent on MailMason.

File: README.md
Comment (derekrushforth) [inline README.md:23]: There are several details in this wiki entry that aren't relevant to grunt-postmark so this might confuse people who aren't using MailMason. 

We should replace this with setup instructions specific to grunt-postmark. I can create the wiki page and link it once we've merged this PR.

File: Gruntfile.js
Comment (derekrushforth) [inline Gruntfile.js]: Optionally storing the template data in a separate JSON file is a good idea, but this might be a roadblock for people who are configuring this for the first time.

`example_config.json` references `templates.json` so I ran into issues during setup because it wasn't clear whether the plugin was using the data from the template config or from the target config.

Instead of reading from `templates.json` we should remove this and document it as an optional step. This would reduce the friction during the initial setup process.

File: tasks/grunt-postmark-templates.js
Comment (derekrushforth) [inline tasks/grunt-postmark-templates.js]: Let's use camel case variables instead of title case. Same goes for `subject`, `htmlBody`, and `textBody` validation. 

I was using the default variables specified from the gruntfile and kept running into validation errors because the variables didn't match.

File: Gruntfile.js
Comment (derekrushforth) [inline Gruntfile.js]: Looks like the upload task is writing the string to the Postmark template instead of reading and posting the contents of the file.

Am I missing something?

screen shot 2017-11-24 at 10 24 21 am

**Review by derekrushforth — CHANGES_REQUESTED** [review]
Hey @randytarampi,

Thanks for taking the time to make this better!

Took a first pass at testing this and ran into a few issues with the default configuration. Take a look at my notes and let me know what you think. I'm also having issues generating the template output file, but I'll hold off on posting those details once we streamline the configuration a bit. 

Happy to jump in and make changes to the codebase if you don't have time. Just let me know!

**Comment by derekrushforth** [comment]
Hey @randytarampi, just checking in to see how this is going. Let me know and I'll help out with some of these code changes.

**Comment by randytarampi** [comment]
Whoops. Just saw this email now. :\

I'll get these fixes done this weekend.

**Comment by randytarampi** [comment]
Hey @derekrushforth, still around, or have you taken off for the holidays yet? Any comments on my last changeset?

**Comment by derekrushforth** [comment]
Hey @randytarampi, sorry to keep you waiting. I'm back from holiday vacation now so I'll take a look at your latest changes.

**Review by derekrushforth — APPROVED** [review]

**Comment by derekrushforth** [comment]
Looking great @randytarampi! Thanks for taking the time to make these changes.

After testing the workflow some more, I think the ideal approach is using the JSON template config right off the bat and using that as the source of truth rather than using the task target config. I know I asked you to remove the logic from the gruntfile, but we should make this super clear in the setup instructions.

Do you want to take a stab at adding instructions to the readme? If you don't have time I can always merge the PR and update the readme separately.

**Comment by derekrushforth** [comment]
Can you also bump the version in package.json too?

**Comment by randytarampi** [comment]
Cool. I'll write things up this weekend.

**Pull request merged**
